test/plugin_functional/plugins/core_config_schema/server/plugin.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { Plugin, CoreSetup } from 'kibana/server';
+
+export class CoreConfigSchemaPlugin implements Plugin {
+  public setup(core: CoreSetup, deps: {}) {
+    const router = core.http.createRouter();
+    router.get(
+      {
+        path: '/api/core_config_schema/foo',
+        validate: {
+          query: schema.object({
+            foo: schema.string(),
+            bar: schema.number(),
+          }),
+        },
```

Comment:
Note: the presence of this plugin is sufficient to test the fix, as all test suites in `test/plugin_functional` would fail when the server loads the plugin. I will remove the plugin when we have other test plugins using route validation, probably in https://github.com/elastic/kibana/pull/63549 as it got one.

Reply:
I'm not sure we need it in the master. This functionality will be covered by the unit-tests and tests in #63549 or updated https://github.com/elastic/kibana/issues/61652

I'm just concerned with many files without a clear intention. We can add your comment to the plugin body, though.

Reply:
Yea, I agree. I was wondering if I should remove it. I initially added it as the issue is not easily reproductible outside of the CI builds, but now that I got a green build, I can remove it if we think this is better. WDYT?

Reply:
👍 let's remove then